### PR TITLE
feat(api): add endpoints for customers

### PR DESCRIPTION
**Pull request by Francesco-code1**

### CODE REVIEW

File: app/routes/api/v1/warehouse/customers/$customerId/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  }
+
+  const customer = await findCustomer(customerId);
+
+  if (customer == null) {
+    throw notFoundResponse();
+  }
+
+  return customer;
+};
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/customers/$customerId/index.ts:23]: All of this is not necessary.

The `id` will be retrieved from the jwt once logged in.

Please refactor this route

File: app/routes/api/v1/warehouse/customers/by-login/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const schema = z.object({
+    email: z.string(),
+    password: z.string(),
+  });
+
+  const data = await parseBody(request, schema);
+  const customer = await findCustomerByLogin(data);
+
+  return customer;
+};
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/customers/by-login/index.ts:23]: This route is not needed

File: app/routes/api/v1/warehouse/customers/creditCard/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }),
+  });
+
+  const data = await parseBody(request, schema);
+  const customer = await createCustomerCreditCard(
+    data.customerId,
+    data.creditCard
+  );
+  return customer;
+};
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/customers/creditCard/index.ts:30]: This must be placed in `app/routes/api/v1/ecommerce`

File: app/routes/api/v1/warehouse/customers/index.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  const schema = z.object({
+    email: z.string(),
+    password: z.string(),
+    firstName: z.string(),
+    lastName: z.string(),
+    bllingAddressId: z.string().nullable(),
+    shippingAddressId: z.string().nullable(),
+  });
+
+  const data = await parseBody(request, schema);
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/customers/index.ts:31]: All of this must be placed in the `ecommerce` route

**Review by nullndr — CHANGES_REQUESTED** [review]
🛑

File: app/routes/api/v1/ecommerce/customers/index.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+export const action: ActionFunction = async ({ request }) => {
+  switch (request.method.toLowerCase()) {
+    case "post":
+      return postRequest(request);
+    case "patch":
+      return patchRequest(request);
+    case "delete":
+      return deleteRequest(request);
+  }
```
Comment (DaruZero) [inline app/routes/api/v1/ecommerce/customers/index.ts:18]:
```
default:
    throw methodNotAllowedResponse();
```

**Review by DaruZero — CHANGES_REQUESTED** [review]
🛑

**Review by DaruZero — APPROVED** [review]
🚀

**Pull request merged**